src/common/components/ui/formik/UploadFieldCertificates.tsx

```diff
@@ -0,0 +1,133 @@
+import * as React from 'react';
+import { FormGroup, FileUpload } from '@patternfly/react-core';
+import { UploadFieldProps } from './types';
+import { getFieldId } from './utils';
+import HelperText from './HelperText';
+import { useTranslation } from '../../../hooks/use-translation-wrapper';
+import { useField } from 'formik';
+
+const UploadField: React.FC<UploadFieldProps> = ({
+  label,
+  labelIcon,
+  helperText,
+  isRequired,
+  children,
+  idPostfix,
+  isDisabled,
+  name,
+}) => {
+  const { t } = useTranslation();
+
+  const maxFileSize = 100000; //100 kb
+  const acceptedFiles =
+    '.pem,.crt,.ca,.cert,application/x-pem-file,application/x-x509-ca-cert,text/plain';
+
+  const fieldId = getFieldId(name, 'input', idPostfix);
+  const [field, , { setValue }] = useField<string | File>(name);
+  const [filename, setFilename] = React.useState('');
+  const [isLoading, setIsLoading] = React.useState(false);
+  const [isRejected, setIsRejected] = React.useState(false);
+  const [errorMessage, setErrorMessage] = React.useState('');
+
+  const handleFileInputChange = (
+    _event: React.ChangeEvent<HTMLInputElement> | React.DragEvent<HTMLElement>,
+    file: File,
+  ) => {
+    setFilename(file.name);
+  };
+
+  const handleTextOrDataChange = (value: string) => {
+    const contentFile = new Blob([value], { type: 'text/plain;charset=utf-8' });
+    if (contentFile.size > maxFileSize) {
+      setIsRejected(true);
+      setErrorMessage(t('ai:The file is too big. Upload a file up to 100 Kb.'));
```

Comment:
I would pass the fileSize to the message, so that if the size limit changes, the translation doesn't need to be updated.

Reply:
True, I've changed it.